src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/acs_client.py

```diff
@@ -12,13 +13,35 @@
 from sshtunnel import SSHTunnelForwarder
 from scp import SCPClient
 
+from azure.cli.core._util import CLIError
 
-def SecureCopy(user, host, src, dest,
-               key_filename=os.path.join(os.path.expanduser("~"), '.ssh', 'id_rsa')):
+
+def _load_key(key_filename, key_password):
+    pkey = None
+    key_pass = None
+    if key_password:
+        key_pass = getpass.getpass()
```

Comment:
Can you use the centralized `azure.cli.core.prompting prompt_pass()` method for this?
This has centralized logic for checking if tty is available.
Then you can choose how to handle the `NoTTYException` in here.